File: GerenciamentoBananas.java
import java.util.ArrayList;
import java.util.List;

public class GerenciamentoBananas {
    private List<Banana> listaBananas;

    public GerenciamentoBananas() {
        this.listaBananas = new ArrayList<>();
    }

    public void cadastrarBanana(Banana banana) {
        listaBananas.add(banana);
    }

    public Banana buscarBanana(int id) {
        for (Banana banana : listaBananas) {
            if (banana.getId() == id) {
                return banana;
            }
        }
        return null;
    }

    public List<Banana> listarBananas() {
        return listaBananas;
    }

    public void atualizarBanana(int id, String nome, String origem, double peso, boolean madura) {
        Banana banana = buscarBanana(id);
        if (banana != null) {
            banana.setNome(nome);
            banana.setOrigem(origem);
            banana.setPeso(peso);
            banana.setMadura(madura);
        }
    }

    public void excluirBanana(int id) {
        Banana banana = buscarBanana(id);
        if (banana != null) {
            listaBananas.remove(banana);
        }
    }
}

File: GerenciamentoBananas.cs
using System;
using System.Collections.Generic;

public class GerenciamentoBananas
{
    private List<Banana> listaBananas;

    public GerenciamentoBananas()
    {
        listaBananas = new List<Banana>();
    }

    public void CadastrarBanana(Banana banana)
    {
        listaBananas.Add(banana);
    }

    public Banana BuscarBanana(int id)
    {
        return listaBananas.Find(b => b.Id == id);
    }

    public List<Banana> ListarBananas()
    {
        return listaBananas;
    }

    public void AtualizarBanana(int id, string nome, string origem, double peso, bool madura)
    {
        Banana banana = BuscarBanana(id);
        if (banana != null)
        {
            banana.Nome = nome;
            banana.Origem = origem;
            banana.Peso = peso;
            banana.Madura = madura;
        }
    }

    public void ExcluirBanana(int id)
    {
        Banana banana = BuscarBanana(id);
        if (banana != null)
        {
            listaBananas.Remove(banana);
        }
    }
}
